examples/controls_example/public/react_controls/types.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+
+import { SerializedStyles } from '@emotion/react';
+import { ControlWidth } from '@kbn/controls-plugin/public/types';
+import { HasSerializableState } from '@kbn/presentation-containers';
+import { PanelCompatibleComponent } from '@kbn/presentation-panel-plugin/public/panel_component/types';
+import {
+  HasParentApi,
+  HasType,
+  HasUniqueId,
+  PublishesBlockingError,
+  PublishesDataLoading,
+  PublishesDisabledActionIds,
+  PublishesPanelTitle,
+  PublishesUnsavedChanges,
+  PublishingSubject,
+  StateComparators,
+} from '@kbn/presentation-publishing';
+
+import { ControlGroupApi } from './control_group/types';
+
+export interface PublishesControlDisplaySettings {
+  grow: PublishingSubject<boolean | undefined>;
+  width: PublishingSubject<ControlWidth | undefined>;
+}
+
+export interface HasCustomPrepend {
+  getCustomPrepend: () => React.FC<{}>;
+}
+
+export type DefaultControlApi = PublishesDataLoading &
+  PublishesBlockingError &
+  PublishesUnsavedChanges &
+  PublishesControlDisplaySettings &
+  Partial<PublishesPanelTitle & PublishesDisabledActionIds & HasCustomPrepend> &
+  HasType &
+  HasUniqueId &
+  HasSerializableState &
+  HasParentApi<ControlGroupApi> & {
+    setDataLoading: (loading: boolean) => void;
+    setBlockingError: (error: Error | undefined) => void;
+  };
+
+export interface DefaultControlState {
```

Comment:
What are the benefits of having each child control manage its own grow and width state rather than the control group managing that for all of the controls? IMO only the control frame / renderer needs to care about this.

Is it because we want to expose this in the editor for the individual control?

Reply:
> Is it because we want to expose this in the editor for the individual control?
Pretty much, yup. It felt cleaner to me than having the control group manage all of this in some sort of child map - and then having to fetch the appropriate value for that specific child in the `DataControlEditor` 🤷 Open to discussion about this, though

Reply:
I see. This is one of those weird in-between types of state. I think what you have works pretty well for now, we'll discuss more during the control group refactor.